### ISSUE TITLE
EAGLE-Qwen2-7B is unable to  get right output in the main branch!

### ISSUE THREAD

**Issue by Elisabethhui** [SafeAILab/EAGLE#162]
def initialize_tree(input_ids, model, past_key_values, logits_processor):
    outputs, orig, hidden_states = model(
        input_ids, past_key_values=past_key_values, output_orig=True
    )

    if logits_processor is not None:
        logits = orig[:, -1]
        logits = logits_processor(None, logits)
        probabilities = torch.nn.functional.softmax(logits, dim=1)
        token = torch.multinomial(probabilities, 1)
    else:
        token = torch.argmax(orig[:, -1])
        token = token[None, None]
    input_ids = torch.cat((input_ids, token.to(input_ids.device)), dim=1)
    # Clone the output hidden states

    draft_tokens, retrieve_indices,tree_mask,tree_position_ids = model.ea_layer.topK_genrate(hidden_states, input_ids, model.base_model.lm_head,logits_processor)
    return draft_tokens, retrieve_indices,tree_mask,tree_position_ids, orig, hidden_states, token
the logits of temperature=0.5 withlogits_processor is always nan, when i change the logits with orig[:,0,:],other errors of draftmodel gennerate is still nan.

**Comment by FuryMartin** [comment]
Hi I encountered the same issue.
 
> When Qwen2 is the target model, please use bf16 precision instead of fp16 to avoid numerical overflow.

Notice this in README.md, I changed the below code from `torch.float16` to `torch.bfloat16` and it works. https://github.com/SafeAILab/EAGLE/blob/59148f5cd49d80aaa13edfeeafaa3d20e381b4ea/eagle/application/webui.py#L275
